src/libsync/configfile.cpp

```diff
@@ -143,6 +145,86 @@ bool copy_dir_recursive(QString from_dir, QString to_dir)
     return true;
 }
 
+// Since QWindow has no saveGeometry, let's roll out our own,
+// this is obviously less smart since we can't query the layout system
+QByteArray saveWindowGeometry(QWindow *w)
+{
+    QByteArray array;
+    QDataStream stream(&array, QIODevice::WriteOnly);
+    stream.setVersion(QDataStream::Qt_5_12);
+    const quint32 magicNumber = 0x430A7465;
+    // Version history:
+    // - Nextcloud Client 3.1.0 : Version 1
+    const quint16 version = 1;
+    const int screenNumber = QGuiApplication::screens().indexOf(w->screen());
+    stream << magicNumber
+           << version
+           << w->geometry()
+           << qint32(screenNumber);
+    return array;
+}
+
+// This is almost a straight copy from Qt code
+static void checkRestoredGeometry(const QRect &availableGeometry, QRect *restoredGeometry)
+{
+    const auto frameHeight = 20;
+
+    if (!restoredGeometry->intersects(availableGeometry)) {
+        restoredGeometry->moveBottom(qMin(restoredGeometry->bottom(), availableGeometry.bottom()));
```

Comment:
@er-vin Can this be achieved with just calling **translate** or **moveToo** method of **QRect**? Also, maybe it's better to just put the window into the center of the screen in case we get into this condition?

Reply:
Yes, but this would actually lead to more verbose code. I used exactly the code used in Qt for the widget variant though.